cloudinit/cmd/main.py

```diff
@@ -320,7 +320,8 @@ def main_init(name, args):
 
     # Stage 5
     try:
-        init.fetch(existing=existing)
+        init.fetch(existing=existing,
+                   fallback=(mode == sources.DSMODE_NETWORK))
```

Comment:
This currently restricts fallback to only NETWORK datasources.  I believe OVF would like this set to fallback=True; which would allow a local-time datasource to fallback to the previous datasource.  

Testing this out, I simulated a dead IMDS by dropping packets heading there:

```
sudo iptables -A OUTPUT -d 169.254.169.254 -j DROP
```
Then I ran
```
cloud-init --debug init --local
cloud-init --debug init
```

The previous object was restored from cache at each stage.